Unable to pass form data to my resource

Hi, I try to make a POST request to a rest resource with this Concourse Resource and I have the following error:
```
http response code: 500
http response text: {"_embedded":{"errors":[{"message":"Error reading from class path resource [\"http://artefact.my-repo.com/artifactory/Maven/appsList/0.0.1/appsList-0.0.1.properties\"]","logref":"RuntimeException","_links":{"self":{"href":"/"}}}]}}
Traceback (most recent call last):
File "/opt/resource/out", line 106, in <module>
print(HTTPResource().run(os.path.basename(__file__), sys.stdin.read(), sys.argv[1:]))
File "/opt/resource/out", line 82, in run
status_code, text = self.cmd(command_argument, rendered_params)
File "/opt/resource/out", line 42, in cmd
raise Exception('Unexpected response {}'.format(response.status_code))
Exception: Unexpected response 500
```

When I try with curl it works perfectly
```
curl --location --request POST 'http://dataflow.my-platform/apps' \
--form 'uri="http://artefact.my-repo.com/artifactory/Maven/appsList/0.0.1/appsList-0.0.1.properties"'
```

Here is my config:
```yaml
resource_types:
- name: http-api
  type: docker-image
  source:
    repository: aequitas/http-api-resource

resources:
- name: dataflow-rest-api
  type: http-api
  source:
    uri: http://dataflow.my-platform

jobs:
- name: load-apps-with-rest-api
  public: true
  plan:
  - put: dataflow-rest-api
    params:
      uri: http://dataflow.my-platform/apps
      method: POST
      debug: true
      form_data:
        uri: http://artefact.my-repo.com/artifactory/Maven/appsList/0.0.1/appsList-0.0.1.properties
```

And [here](https://docs.spring.io/spring-cloud-dataflow/docs/current/reference/htmlsingle/#resources-app-registry-bulk), the API I try to call.

I don't see what I did wrong... Isn't that the way to pass form-data arguments?

Here is the whole trace
```
$ docker run --rm aequitas/http-resource-test
============================= test session starts ==============================
platform linux -- Python 3.9.2, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
rootdir: /opt/resource-tests
plugins: httpbin-1.0.0, pylama-7.7.1
collected 8 items

opt/resource-tests/test_hipchat.py s                                     [ 12%]
opt/resource-tests/test_invocation.py FFFFF.F                            [100%]

=================================== FAILURES ===================================
___________________________________ test_out ___________________________________
opt/resource-tests/test_invocation.py:18: in test_out
    subprocess.check_output('/opt/resource/out', input=json.dumps(data).encode())
        data       = {'source': {'uri': 'http://127.0.0.1:45557/status/200'}, 'version': {}}
        httpbin    = <pytest_httpbin.serve.Server object at 0x7fda761698e0>
usr/local/lib/python3.9/subprocess.py:424: in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
        kwargs     = {'input': b'{"source": {"uri": "http://127.0.0.1:45557/status/200"}, "version": {}}'}
        popenargs  = ('/opt/resource/out',)
        timeout    = None
usr/local/lib/python3.9/subprocess.py:505: in run
    with Popen(*popenargs, **kwargs) as process:
        capture_output = False
        check      = True
        input      = b'{"source": {"uri": "http://127.0.0.1:45557/status/200"}, "version": {}}'
        kwargs     = {'stdin': -1, 'stdout': -1}
        popenargs  = ('/opt/resource/out',)
        timeout    = None
usr/local/lib/python3.9/subprocess.py:951: in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
        args       = '/opt/resource/out'
        bufsize    = -1
        c2pread    = 14
        c2pwrite   = 15
        close_fds  = True
        creationflags = 0
        cwd        = None
        encoding   = None
        env        = None
        errors     = None
        errread    = -1
        errwrite   = -1
        executable = None
        extra_groups = None
        f          = <_io.BufferedReader name=14>
        gid        = None
        gids       = None
        group      = None
        p2cread    = 12
        p2cwrite   = 13
        pass_fds   = ()
        preexec_fn = None
        restore_signals = True
        self       = <Popen: returncode: 255 args: ['/', 'o', 'p', 't', '/', 'r', 'e', 's', 'o', ...>
        shell      = False
        start_new_session = False
        startupinfo = None
        stderr     = None
        stdin      = -1
        stdout     = -1
        text       = None
        uid        = None
        umask      = -1
        universal_newlines = None
        user       = None
usr/local/lib/python3.9/subprocess.py:1823: in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
E   OSError: [Errno 8] Exec format error: '/opt/resource/out'
        args       = ['/opt/resource/out']
        c2pread    = 14
        c2pwrite   = 15
        child_exception_type = <class 'OSError'>
        child_exec_never_called = False
        close_fds  = True
        creationflags = 0
        cwd        = None
        env        = None
        env_list   = None
        err_filename = '/opt/resource/out'
        err_msg    = 'Exec format error'
        errno_num  = 8
        errpipe_data = bytearray(b'OSError:8:')
        errpipe_read = 16
        errpipe_write = 17
        errread    = -1
        errwrite   = -1
        exception_name = bytearray(b'OSError')
        executable = b'/opt/resource/out'
        executable_list = (b'/opt/resource/out',)
        fds_to_keep = {17}
        gid        = None
        gids       = None
        hex_errno  = bytearray(b'8')
        low_fds_to_close = []
        orig_executable = '/opt/resource/out'
        p2cread    = 12
        p2cwrite   = 13
        part       = b''
        pass_fds   = ()
        pid        = 8
        preexec_fn = None
        restore_signals = True
        self       = <Popen: returncode: 255 args: ['/', 'o', 'p', 't', '/', 'r', 'e', 's', 'o', ...>
        shell      = False
        start_new_session = False
        startupinfo = None
        sts        = 65280
        uid        = None
        umask      = -1
_______________________________ test_out_failure _______________________________
opt/resource-tests/test_invocation.py:31: in test_out_failure
    subprocess.check_output('/opt/resource/out', input=json.dumps(data).encode())
        data       = {'source': {'uri': 'http://127.0.0.1:45557/status/404'}, 'version': {}}
        httpbin    = <pytest_httpbin.serve.Server object at 0x7fda761698e0>
usr/local/lib/python3.9/subprocess.py:424: in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
        kwargs     = {'input': b'{"source": {"uri": "http://127.0.0.1:45557/status/404"}, "version": {}}'}
        popenargs  = ('/opt/resource/out',)
        timeout    = None
usr/local/lib/python3.9/subprocess.py:505: in run
    with Popen(*popenargs, **kwargs) as process:
        capture_output = False
        check      = True
        input      = b'{"source": {"uri": "http://127.0.0.1:45557/status/404"}, "version": {}}'
        kwargs     = {'stdin': -1, 'stdout': -1}
        popenargs  = ('/opt/resource/out',)
        timeout    = None
usr/local/lib/python3.9/subprocess.py:951: in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
        args       = '/opt/resource/out'
        bufsize    = -1
        c2pread    = 14
        c2pwrite   = 15
        close_fds  = True
        creationflags = 0
        cwd        = None
        encoding   = None
        env        = None
        errors     = None
        errread    = -1
        errwrite   = -1
        executable = None
        extra_groups = None
        f          = <_io.BufferedReader name=14>
        gid        = None
        gids       = None
        group      = None
        p2cread    = 12
        p2cwrite   = 13
        pass_fds   = ()
        preexec_fn = None
        restore_signals = True
        self       = <Popen: returncode: 255 args: ['/', 'o', 'p', 't', '/', 'r', 'e', 's', 'o', ...>
        shell      = False
        start_new_session = False
        startupinfo = None
        stderr     = None
        stdin      = -1
        stdout     = -1
        text       = None
        uid        = None
        umask      = -1
        universal_newlines = None
        user       = None
usr/local/lib/python3.9/subprocess.py:1823: in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
E   OSError: [Errno 8] Exec format error: '/opt/resource/out'
        args       = ['/opt/resource/out']
        c2pread    = 14
        c2pwrite   = 15
        child_exception_type = <class 'OSError'>
        child_exec_never_called = False
        close_fds  = True
        creationflags = 0
        cwd        = None
        env        = None
        env_list   = None
        err_filename = '/opt/resource/out'
        err_msg    = 'Exec format error'
        errno_num  = 8
        errpipe_data = bytearray(b'OSError:8:')
        errpipe_read = 16
        errpipe_write = 17
        errread    = -1
        errwrite   = -1
        exception_name = bytearray(b'OSError')
        executable = b'/opt/resource/out'
        executable_list = (b'/opt/resource/out',)
        fds_to_keep = {17}
        gid        = None
        gids       = None
        hex_errno  = bytearray(b'8')
        low_fds_to_close = []
        orig_executable = '/opt/resource/out'
        p2cread    = 12
        p2cwrite   = 13
        part       = b''
        pass_fds   = ()
        pid        = 9
        preexec_fn = None
        restore_signals = True
        self       = <Popen: returncode: 255 args: ['/', 'o', 'p', 't', '/', 'r', 'e', 's', 'o', ...>
        shell      = False
        start_new_session = False
        startupinfo = None
        sts        = 65280
        uid        = None
        umask      = -1
__________________________________ test_auth ___________________________________
opt/resource-tests/test_invocation.py:42: in test_auth
    subprocess.check_output('/opt/resource/out', input=json.dumps(data).encode())
        data       = {'source': {'uri': 'http://user:password@127.0.0.1:45557/basic-auth/user/password'}}
        httpbin    = <pytest_httpbin.serve.Server object at 0x7fda761698e0>
usr/local/lib/python3.9/subprocess.py:424: in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
        kwargs     = {'input': b'{"source": {"uri": "http://user:password@127.0.0.1:45557/basic-auth/user/password"}}'}
        popenargs  = ('/opt/resource/out',)
        timeout    = None
usr/local/lib/python3.9/subprocess.py:505: in run
    with Popen(*popenargs, **kwargs) as process:
        capture_output = False
        check      = True
        input      = b'{"source": {"uri": "http://user:password@127.0.0.1:45557/basic-auth/user/password"}}'
        kwargs     = {'stdin': -1, 'stdout': -1}
        popenargs  = ('/opt/resource/out',)
        timeout    = None
usr/local/lib/python3.9/subprocess.py:951: in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
        args       = '/opt/resource/out'
        bufsize    = -1
        c2pread    = 14
        c2pwrite   = 15
        close_fds  = True
        creationflags = 0
        cwd        = None
        encoding   = None
        env        = None
        errors     = None
        errread    = -1
        errwrite   = -1
        executable = None
        extra_groups = None
        f          = <_io.BufferedReader name=14>
        gid        = None
        gids       = None
        group      = None
        p2cread    = 12
        p2cwrite   = 13
        pass_fds   = ()
        preexec_fn = None
        restore_signals = True
        self       = <Popen: returncode: 255 args: ['/', 'o', 'p', 't', '/', 'r', 'e', 's', 'o', ...>
        shell      = False
        start_new_session = False
        startupinfo = None
        stderr     = None
        stdin      = -1
        stdout     = -1
        text       = None
        uid        = None
        umask      = -1
        universal_newlines = None
        user       = None
usr/local/lib/python3.9/subprocess.py:1823: in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
E   OSError: [Errno 8] Exec format error: '/opt/resource/out'
        args       = ['/opt/resource/out']
        c2pread    = 14
        c2pwrite   = 15
        child_exception_type = <class 'OSError'>
        child_exec_never_called = False
        close_fds  = True
        creationflags = 0
        cwd        = None
        env        = None
        env_list   = None
        err_filename = '/opt/resource/out'
        err_msg    = 'Exec format error'
        errno_num  = 8
        errpipe_data = bytearray(b'OSError:8:')
        errpipe_read = 16
        errpipe_write = 17
        errread    = -1
        errwrite   = -1
        exception_name = bytearray(b'OSError')
        executable = b'/opt/resource/out'
        executable_list = (b'/opt/resource/out',)
        fds_to_keep = {17}
        gid        = None
        gids       = None
        hex_errno  = bytearray(b'8')
        low_fds_to_close = []
        orig_executable = '/opt/resource/out'
        p2cread    = 12
        p2cwrite   = 13
        part       = b''
        pass_fds   = ()
        pid        = 10
        preexec_fn = None
        restore_signals = True
        self       = <Popen: returncode: 255 args: ['/', 'o', 'p', 't', '/', 'r', 'e', 's', 'o', ...>
        shell      = False
        start_new_session = False
        startupinfo = None
        sts        = 65280
        uid        = None
        umask      = -1
__________________________________ test_json ___________________________________
opt/resource-tests/test_invocation.py:57: in test_json
    output = cmd('out', source)
        httpbin    = <pytest_httpbin.serve.Server object at 0x7fda761698e0>
        source     = {'json': {'test': 123}, 'method': 'POST', 'uri': 'http://127.0.0.1:45557/post', 'version': {}}
opt/resource-tests/helpers.py:24: in cmd
    output = subprocess.check_output(command, env=environment,
        args       = []
        cmd_name   = 'out'
        command    = ['/opt/resource/out']
        environment = {'BUILD_ID': '123', 'BUILD_JOB_NAME': 'test-job', 'BUILD_NAME': '1', 'BUILD_PIPELINE_NAME': 'test-pipeline', ...}
        in_json    = '{"source": {"uri": "http://127.0.0.1:45557/post", "method": "POST", "json": {"test": 123}, "version": {}}, "version": {}, "params": {}}'
        params     = {}
        source     = {'json': {'test': 123}, 'method': 'POST', 'uri': 'http://127.0.0.1:45557/post', 'version': {}}
        version    = {}
usr/local/lib/python3.9/subprocess.py:424: in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
        kwargs     = {'env': {'BUILD_ID': '123', 'BUILD_JOB_NAME': 'test-job', 'BUILD_NAME': '1', 'BUILD_PIPELINE_NAME': 'test-pipeline', ....ms": {}}', 'stderr': <_io.TextIOWrapper name="<_io.FileIO name=8 mode='rb+' closefd=True>" mode='r+' encoding='utf-8'>}
        popenargs  = (['/opt/resource/out'],)
        timeout    = None
usr/local/lib/python3.9/subprocess.py:505: in run
    with Popen(*popenargs, **kwargs) as process:
        capture_output = False
        check      = True
        input      = b'{"source": {"uri": "http://127.0.0.1:45557/post", "method": "POST", "json": {"test": 123}, "version": {}}, "version": {}, "params": {}}'
        kwargs     = {'env': {'BUILD_ID': '123', 'BUILD_JOB_NAME': 'test-job', 'BUILD_NAME': '1', 'BUILD_PIPELINE_NAME': 'test-pipeline', ....extIOWrapper name="<_io.FileIO name=8 mode='rb+' closefd=True>" mode='r+' encoding='utf-8'>, 'stdin': -1, 'stdout': -1}
        popenargs  = (['/opt/resource/out'],)
        timeout    = None
usr/local/lib/python3.9/subprocess.py:951: in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
        args       = ['/opt/resource/out']
        bufsize    = -1
        c2pread    = 14
        c2pwrite   = 15
        close_fds  = True
        creationflags = 0
        cwd        = None
        encoding   = None
        env        = {'BUILD_ID': '123', 'BUILD_JOB_NAME': 'test-job', 'BUILD_NAME': '1', 'BUILD_PIPELINE_NAME': 'test-pipeline', ...}
        errors     = None
        errread    = -1
        errwrite   = 8
        executable = None
        extra_groups = None
        f          = <_io.BufferedReader name=14>
        gid        = None
        gids       = None
        group      = None
        p2cread    = 12
        p2cwrite   = 13
        pass_fds   = ()
        preexec_fn = None
        restore_signals = True
        self       = <Popen: returncode: 255 args: ['/opt/resource/out']>
        shell      = False
        start_new_session = False
        startupinfo = None
        stderr     = <_io.TextIOWrapper name="<_io.FileIO name=8 mode='rb+' closefd=True>" mode='r+' encoding='utf-8'>
        stdin      = -1
        stdout     = -1
        text       = None
        uid        = None
        umask      = -1
        universal_newlines = None
        user       = None
usr/local/lib/python3.9/subprocess.py:1823: in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
E   OSError: [Errno 8] Exec format error: '/opt/resource/out'
        args       = ['/opt/resource/out']
        c2pread    = 14
        c2pwrite   = 15
        child_exception_type = <class 'OSError'>
        child_exec_never_called = False
        close_fds  = True
        creationflags = 0
        cwd        = None
        env        = {'BUILD_ID': '123', 'BUILD_JOB_NAME': 'test-job', 'BUILD_NAME': '1', 'BUILD_PIPELINE_NAME': 'test-pipeline', ...}
        env_list   = [b'RESOURCE_DEBUG=1', b'HOSTNAME=ddeb6fae8593', b'PYTHON_PIP_VERSION=21.0.1', b'HOME=/root', b'GPG_KEY=E3FF2839C048B25...8', b'PYTHON_GET_PIP_URL=https://github.com/pypa/get-pip/raw/b60e2320d9e8d02348525bd74e871e466afdf77c/get-pip.py', ...]
        err_filename = '/opt/resource/out'
        err_msg    = 'Exec format error'
        errno_num  = 8
        errpipe_data = bytearray(b'OSError:8:')
        errpipe_read = 16
        errpipe_write = 17
        errread    = -1
        errwrite   = 8
        exception_name = bytearray(b'OSError')
        executable = b'/opt/resource/out'
        executable_list = (b'/opt/resource/out',)
        fds_to_keep = {17}
        gid        = None
        gids       = None
        hex_errno  = bytearray(b'8')
        k          = b'TEST'
        low_fds_to_close = []
        orig_executable = '/opt/resource/out'
        p2cread    = 12
        p2cwrite   = 13
        part       = b''
        pass_fds   = ()
        pid        = 11
        preexec_fn = None
        restore_signals = True
        self       = <Popen: returncode: 255 args: ['/opt/resource/out']>
        shell      = False
        start_new_session = False
        startupinfo = None
        sts        = 65280
        uid        = None
        umask      = -1
        v          = 'true'
______________________________ test_interpolation ______________________________
opt/resource-tests/test_invocation.py:79: in test_interpolation
    output = cmd('out', source)
        httpbin    = <pytest_httpbin.serve.Server object at 0x7fda761698e0>
        source     = {'json': {'array': ['{BUILD_NAME}'], 'object': {'test': '{BUILD_NAME}'}}, 'method': 'POST', 'uri': 'http://127.0.0.1:45557/post'}
opt/resource-tests/helpers.py:24: in cmd
    output = subprocess.check_output(command, env=environment,
        args       = []
        cmd_name   = 'out'
        command    = ['/opt/resource/out']
        environment = {'BUILD_ID': '123', 'BUILD_JOB_NAME': 'test-job', 'BUILD_NAME': '1', 'BUILD_PIPELINE_NAME': 'test-pipeline', ...}
        in_json    = '{"source": {"uri": "http://127.0.0.1:45557/post", "method": "POST", "json": {"object": {"test": "{BUILD_NAME}"}, "array": ["{BUILD_NAME}"]}}, "version": {}, "params": {}}'
        params     = {}
        source     = {'json': {'array': ['{BUILD_NAME}'], 'object': {'test': '{BUILD_NAME}'}}, 'method': 'POST', 'uri': 'http://127.0.0.1:45557/post'}
        version    = {}
usr/local/lib/python3.9/subprocess.py:424: in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
        kwargs     = {'env': {'BUILD_ID': '123', 'BUILD_JOB_NAME': 'test-job', 'BUILD_NAME': '1', 'BUILD_PIPELINE_NAME': 'test-pipeline', ....ms": {}}', 'stderr': <_io.TextIOWrapper name="<_io.FileIO name=8 mode='rb+' closefd=True>" mode='r+' encoding='utf-8'>}
        popenargs  = (['/opt/resource/out'],)
        timeout    = None
usr/local/lib/python3.9/subprocess.py:505: in run
    with Popen(*popenargs, **kwargs) as process:
        capture_output = False
        check      = True
        input      = b'{"source": {"uri": "http://127.0.0.1:45557/post", "method": "POST", "json": {"object": {"test": "{BUILD_NAME}"}, "array": ["{BUILD_NAME}"]}}, "version": {}, "params": {}}'
        kwargs     = {'env': {'BUILD_ID': '123', 'BUILD_JOB_NAME': 'test-job', 'BUILD_NAME': '1', 'BUILD_PIPELINE_NAME': 'test-pipeline', ....extIOWrapper name="<_io.FileIO name=8 mode='rb+' closefd=True>" mode='r+' encoding='utf-8'>, 'stdin': -1, 'stdout': -1}
        popenargs  = (['/opt/resource/out'],)
        timeout    = None
usr/local/lib/python3.9/subprocess.py:951: in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
        args       = ['/opt/resource/out']
        bufsize    = -1
        c2pread    = 14
        c2pwrite   = 15
        close_fds  = True
        creationflags = 0
        cwd        = None
        encoding   = None
        env        = {'BUILD_ID': '123', 'BUILD_JOB_NAME': 'test-job', 'BUILD_NAME': '1', 'BUILD_PIPELINE_NAME': 'test-pipeline', ...}
        errors     = None
        errread    = -1
        errwrite   = 8
        executable = None
        extra_groups = None
        f          = <_io.BufferedReader name=14>
        gid        = None
        gids       = None
        group      = None
        p2cread    = 12
        p2cwrite   = 13
        pass_fds   = ()
        preexec_fn = None
        restore_signals = True
        self       = <Popen: returncode: 255 args: ['/opt/resource/out']>
        shell      = False
        start_new_session = False
        startupinfo = None
        stderr     = <_io.TextIOWrapper name="<_io.FileIO name=8 mode='rb+' closefd=True>" mode='r+' encoding='utf-8'>
        stdin      = -1
        stdout     = -1
        text       = None
        uid        = None
        umask      = -1
        universal_newlines = None
        user       = None
usr/local/lib/python3.9/subprocess.py:1823: in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
E   OSError: [Errno 8] Exec format error: '/opt/resource/out'
        args       = ['/opt/resource/out']
        c2pread    = 14
        c2pwrite   = 15
        child_exception_type = <class 'OSError'>
        child_exec_never_called = False
        close_fds  = True
        creationflags = 0
        cwd        = None
        env        = {'BUILD_ID': '123', 'BUILD_JOB_NAME': 'test-job', 'BUILD_NAME': '1', 'BUILD_PIPELINE_NAME': 'test-pipeline', ...}
        env_list   = [b'RESOURCE_DEBUG=1', b'HOSTNAME=ddeb6fae8593', b'PYTHON_PIP_VERSION=21.0.1', b'HOME=/root', b'GPG_KEY=E3FF2839C048B25...8', b'PYTHON_GET_PIP_URL=https://github.com/pypa/get-pip/raw/b60e2320d9e8d02348525bd74e871e466afdf77c/get-pip.py', ...]
        err_filename = '/opt/resource/out'
        err_msg    = 'Exec format error'
        errno_num  = 8
        errpipe_data = bytearray(b'OSError:8:')
        errpipe_read = 16
        errpipe_write = 17
        errread    = -1
        errwrite   = 8
        exception_name = bytearray(b'OSError')
        executable = b'/opt/resource/out'
        executable_list = (b'/opt/resource/out',)
        fds_to_keep = {17}
        gid        = None
        gids       = None
        hex_errno  = bytearray(b'8')
        k          = b'TEST'
        low_fds_to_close = []
        orig_executable = '/opt/resource/out'
        p2cread    = 12
        p2cwrite   = 13
        part       = b''
        pass_fds   = ()
        pid        = 12
        preexec_fn = None
        restore_signals = True
        self       = <Popen: returncode: 255 args: ['/opt/resource/out']>
        shell      = False
        start_new_session = False
        startupinfo = None
        sts        = 65280
        uid        = None
        umask      = -1
        v          = 'true'
_____________________________ test_data_urlencode ______________________________
opt/resource-tests/test_invocation.py:112: in test_data_urlencode
    output = cmd('out', source)
        httpbin    = <pytest_httpbin.serve.Server object at 0x7fda761698e0>
        source     = {'form_data': {'field': {'test': 123}}, 'method': 'POST', 'uri': 'http://127.0.0.1:45557/post'}
opt/resource-tests/helpers.py:24: in cmd
    output = subprocess.check_output(command, env=environment,
        args       = []
        cmd_name   = 'out'
        command    = ['/opt/resource/out']
        environment = {'BUILD_ID': '123', 'BUILD_JOB_NAME': 'test-job', 'BUILD_NAME': '1', 'BUILD_PIPELINE_NAME': 'test-pipeline', ...}
        in_json    = '{"source": {"uri": "http://127.0.0.1:45557/post", "method": "POST", "form_data": {"field": {"test": 123}}}, "version": {}, "params": {}}'
        params     = {}
        source     = {'form_data': {'field': {'test': 123}}, 'method': 'POST', 'uri': 'http://127.0.0.1:45557/post'}
        version    = {}
usr/local/lib/python3.9/subprocess.py:424: in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
        kwargs     = {'env': {'BUILD_ID': '123', 'BUILD_JOB_NAME': 'test-job', 'BUILD_NAME': '1', 'BUILD_PIPELINE_NAME': 'test-pipeline', ....ms": {}}', 'stderr': <_io.TextIOWrapper name="<_io.FileIO name=8 mode='rb+' closefd=True>" mode='r+' encoding='utf-8'>}
        popenargs  = (['/opt/resource/out'],)
        timeout    = None
usr/local/lib/python3.9/subprocess.py:505: in run
    with Popen(*popenargs, **kwargs) as process:
        capture_output = False
        check      = True
        input      = b'{"source": {"uri": "http://127.0.0.1:45557/post", "method": "POST", "form_data": {"field": {"test": 123}}}, "version": {}, "params": {}}'
        kwargs     = {'env': {'BUILD_ID': '123', 'BUILD_JOB_NAME': 'test-job', 'BUILD_NAME': '1', 'BUILD_PIPELINE_NAME': 'test-pipeline', ....extIOWrapper name="<_io.FileIO name=8 mode='rb+' closefd=True>" mode='r+' encoding='utf-8'>, 'stdin': -1, 'stdout': -1}
        popenargs  = (['/opt/resource/out'],)
        timeout    = None
usr/local/lib/python3.9/subprocess.py:951: in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
        args       = ['/opt/resource/out']
        bufsize    = -1
        c2pread    = 14
        c2pwrite   = 15
        close_fds  = True
        creationflags = 0
        cwd        = None
        encoding   = None
        env        = {'BUILD_ID': '123', 'BUILD_JOB_NAME': 'test-job', 'BUILD_NAME': '1', 'BUILD_PIPELINE_NAME': 'test-pipeline', ...}
        errors     = None
        errread    = -1
        errwrite   = 8
        executable = None
        extra_groups = None
        f          = <_io.BufferedReader name=14>
        gid        = None
        gids       = None
        group      = None
        p2cread    = 12
        p2cwrite   = 13
        pass_fds   = ()
        preexec_fn = None
        restore_signals = True
        self       = <Popen: returncode: 255 args: ['/opt/resource/out']>
        shell      = False
        start_new_session = False
        startupinfo = None
        stderr     = <_io.TextIOWrapper name="<_io.FileIO name=8 mode='rb+' closefd=True>" mode='r+' encoding='utf-8'>
        stdin      = -1
        stdout     = -1
        text       = None
        uid        = None
        umask      = -1
        universal_newlines = None
        user       = None
usr/local/lib/python3.9/subprocess.py:1823: in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
E   OSError: [Errno 8] Exec format error: '/opt/resource/out'
        args       = ['/opt/resource/out']
        c2pread    = 14
        c2pwrite   = 15
        child_exception_type = <class 'OSError'>
        child_exec_never_called = False
        close_fds  = True
        creationflags = 0
        cwd        = None
        env        = {'BUILD_ID': '123', 'BUILD_JOB_NAME': 'test-job', 'BUILD_NAME': '1', 'BUILD_PIPELINE_NAME': 'test-pipeline', ...}
        env_list   = [b'RESOURCE_DEBUG=1', b'HOSTNAME=ddeb6fae8593', b'PYTHON_PIP_VERSION=21.0.1', b'HOME=/root', b'GPG_KEY=E3FF2839C048B25...8', b'PYTHON_GET_PIP_URL=https://github.com/pypa/get-pip/raw/b60e2320d9e8d02348525bd74e871e466afdf77c/get-pip.py', ...]
        err_filename = '/opt/resource/out'
        err_msg    = 'Exec format error'
        errno_num  = 8
        errpipe_data = bytearray(b'OSError:8:')
        errpipe_read = 16
        errpipe_write = 17
        errread    = -1
        errwrite   = 8
        exception_name = bytearray(b'OSError')
        executable = b'/opt/resource/out'
        executable_list = (b'/opt/resource/out',)
        fds_to_keep = {17}
        gid        = None
        gids       = None
        hex_errno  = bytearray(b'8')
        k          = b'TEST'
        low_fds_to_close = []
        orig_executable = '/opt/resource/out'
        p2cread    = 12
        p2cwrite   = 13
        part       = b''
        pass_fds   = ()
        pid        = 14
        preexec_fn = None
        restore_signals = True
        self       = <Popen: returncode: 255 args: ['/opt/resource/out']>
        shell      = False
        start_new_session = False
        startupinfo = None
        sts        = 65280
        uid        = None
        umask      = -1
        v          = 'true'
=========================== short test summary info ============================
FAILED opt/resource-tests/test_invocation.py::test_out - OSError: [Errno 8] E...
FAILED opt/resource-tests/test_invocation.py::test_out_failure - OSError: [Er...
FAILED opt/resource-tests/test_invocation.py::test_auth - OSError: [Errno 8] ...
FAILED opt/resource-tests/test_invocation.py::test_json - OSError: [Errno 8] ...
FAILED opt/resource-tests/test_invocation.py::test_interpolation - OSError: [...
FAILED opt/resource-tests/test_invocation.py::test_data_urlencode - OSError: ...
==================== 6 failed, 1 passed, 1 skipped in 1.05s ====================
```

--- a/README.md
+++ b/README.md
@@ -38,6 +38,8 @@ Options set in `params` take precedence over options in `source`.
 
 * `form_data`: *Optional* Dictionary with form field/value pairs to send as data. Values are converted to JSON and URL-encoded.
 
+* `parse_form_data`: *Optional* Boolean to specify if form_data should be converted to JSON or not (default `true`). If false, set `"Content-Type":"application/x-www-form-urlencoded"` header.
+
 ## Behavior
 
 Currently the only useful action the resource supports is `out`. The actions `in` and `check` will be added later.
@@ -129,4 +131,32 @@ jobs:
 
           - put: jenkins-trigger-job
 
+```
+
+
+### Call a Rest API with data formated in a standard form
+
+Register apps on a Spring Cloud Data Flow instance via a `POST` formated with `application/x-www-form-urlencoded` encoding type.
+
+More info: https://docs.spring.io/spring-cloud-dataflow/docs/current/reference/htmlsingle/#resources-app-registry-bulk
+
+```yaml
+resources:
+- name: dataflow-rest-api
+  type: http-api
+  source:
+    uri: http://my-dataflow-platform.org
+
+jobs:
+- name: load-apps
+  public: true
+  plan:
+  - put: dataflow-rest-api
+    params:
+      uri: http://my-dataflow-platform.org/apps
+      method: POST
+      debug: true
+      parse_form_data: false
+      form_data:
+        uri: http://artefact.my-repo.com/artifactory/Maven/appsList/0.0.1/appsList-0.0.1.properties
 ```

--- a/assets/resource.py
+++ b/assets/resource.py
@@ -22,6 +22,7 @@ class HTTPResource:
         ssl_verify = data.get('ssl_verify', True)
         ok_responses = data.get('ok_responses', [200, 201, 202, 204])
         form_data = data.get('form_data')
+        parse_form_data = data.get('parse_form_data', True)
 
         if isinstance(ssl_verify, bool):
             verify = ssl_verify
@@ -29,8 +30,10 @@ class HTTPResource:
             verify = str(tempfile.NamedTemporaryFile(delete=False, prefix='ssl-').write(verify))
 
         request_data = None
-        if form_data:
+        if form_data and parse_form_data:
             request_data = {k: json.dumps(v, ensure_ascii=False) for k, v in form_data.items()}
+        elif form_data and not parse_form_data:
+            request_data = form_data
 
         response = requests.request(method, uri, json=json_data,
                                     data=request_data, headers=headers, verify=verify)

--- a/test/test_invocation.py
+++ b/test/test_invocation.py
@@ -131,3 +131,22 @@ def test_data_ensure_ascii(httpbin):
     output = cmd('out', source)
 
     assert output['form'] == {'field': '{"test": "日本語"}'}
+
+
+def test_not_parsed_data(httpbin):
+    """Test form_data in a standard format."""
+
+    source = {
+        'uri': httpbin + '/post',
+        'method': 'POST',
+        'parse_form_data': False,
+        'form_data': {
+            'firstname': 'John',
+            'lastname': 'Doe'
+        }
+    }
+
+    output = cmd('out', source)
+
+    assert output['form'] == {"firstname":"John","lastname":"Doe"}
+    assert output['version'] == {}

